### PR TITLE
Add `tuple` argument to `ip_network`

### PR DESCRIPTION
**Pull request by sobolevn**
Source: 
- https://github.com/python/cpython/blob/6fbd889d6e937ad255f98b5495b78a06d05640d5/Lib/ipaddress.py#L2090
- https://github.com/python/cpython/blob/6fbd889d6e937ad255f98b5495b78a06d05640d5/Lib/ipaddress.py#L534-L555

Closes https://github.com/python/typeshed/issues/8570

### CODE REVIEW

File: stdlib/ipaddress.pyi [hunk 1]
```diff
@@ -15,7 +15,7 @@ _RawIPAddress: TypeAlias = int | str | bytes | IPv4Address | IPv6Address
 _RawNetworkPart: TypeAlias = IPv4Network | IPv6Network | IPv4Interface | IPv6Interface
 
 def ip_address(address: _RawIPAddress) -> IPv4Address | IPv6Address: ...
-def ip_network(address: _RawIPAddress | _RawNetworkPart, strict: bool = ...) -> IPv4Network | IPv6Network: ...
+def ip_network(address: _RawIPAddress | _RawNetworkPart | Tuple[_RawIPAddress], Tuple[_RawIPAddress, int], strict: bool = ...) -> IPv4Network | IPv6Network: ...
```
Comment (AlexWaygood) [inline stdlib/ipaddress.pyi:18]:
```suggestion
def ip_network(address: _RawIPAddress | _RawNetworkPart | tuple[_RawIPAddress] | tuple[_RawIPAddress, int], strict: bool = ...) -> IPv4Network | IPv6Network: ...
```

**Review by AlexWaygood — COMMENTED** [review]
I'm a bit uneasy about this change. It's true that it works at runtime, but passing a `tuple` in here seems to be entirely undocumented and untested, so I feel like it could change at any time without any backwards-compatibility guarantees.

**Comment by sobolevn** [comment]
I don't have any opinion really

**Comment by Akuli** [comment]
We should in general type things as the implementation does it, not as documented. I think it applies here too. I also don't see any reason why tests for this function should check with all types that `IPv4Network` and `IPv6Network` accept, even though that's the intention; that would make the tests longer without increasing the test coverage.

If someone still feels "uneasy" about this, we could add a test case -- again, nothing too thorough, could be for example the two examples in the issue.

**Comment by Akuli** [comment]
And I just realized that adding a test case to typeshed would be pointless, as those don't run at runtime :) I still think we should support this, because it works.

**Comment by AlexWaygood** [comment]
> And I just realized that adding a test case to typeshed would be pointless, as those don't run at runtime :) I still think we should support this, because it works.

I'm fine with this being merged, just don't necessarily want to be the one to hit merge myself :)

I agree that in general, we should follow the implementation rather than the documentation.

**Comment by Akuli** [comment]
Passing in a tuple seems to be a common pattern: https://grep.app/search?q=ip_network%28%28

**Review by AlexWaygood — APPROVED** [review]
I'm convinced.

**Comment by sobolevn** [comment]
🥳

**Pull request merged**